src/rulesets/Base/array.jl

```diff
@@ -103,3 +103,40 @@ function rrule(::typeof(fill), value::Any, dims::Int...)
     end
     return fill(value, dims), fill_pullback
 end
+
+#####
+##### `repeat`
+#####
+
+function rrule(::typeof(repeat), x::AbstractVector, m::Integer)
+    function repeat_pullback(Ȳ)
+        return (NO_FIELDS, dropdims(sum(reshape(Ȳ, length(x), :); dims=2); dims=2), DoesNotExist())
```

Comment:
should this close over `length(x)` rather than `x` (to reduce memory requirements)?
```suggestion
    length_x = length(x)
    function repeat_pullback(Ȳ)
        return (NO_FIELDS, dropdims(sum(reshape(Ȳ, length_x, :); dims=2); dims=2), DoesNotExist())
```
And likewise with the `size` calls in the rules below. 

This is a genuine question, i'm not sure what best practice here is (cc @oxinabox).

I believe this would be manually doing one thing we want an `OpaqueClosure` to be able to do automatically. 
Also maybe this should be mentioned on the "writing good rules docs"?

Reply:
Yeah, I am cautious about doing that since it makes code uglier.
So I wouldn't block a PR over it (it can always be optimized later)
But it does save memory.
We should discuss this with pros and cons in "Writing God Rules"